pkgs/tools/X11/xdg-utils/default.nix

```diff
@@ -15,73 +16,233 @@ let
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes ++ [
+        "gconftool-2"
+        "xdg-user-dir"
+      ];
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-desktop-menu" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk ];
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-email" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk "${placeholder "out"}/bin" ];
+      execer = [
+        "cannot:${placeholder "out"}/bin/xdg-mime"
+        "cannot:${placeholder "out"}/bin/xdg-open"
+      ];
+      fake.external = commonFakes ++ [
+        "exo-open"
+        "gconftool-2"
+        "gdbus"
+        "gio"
+        "gnome-open"
+        "gvfs-open"
+        "qtxdg-mat"
+        "xdg-email-hook.sh"
+      ];
+      fix."/bin/echo" = true;
+      keep = {
+        "$command" = true;
+        "$kreadconfig" = true;
+        "$THUNDERBIRD" = true;
+        "$utf8" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-icon-resource" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-mime" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ file gawk ];
+      fake.external = commonFakes ++ [
+        "gio"
+        "gnomevfs-info"
+        "gvfs-info"
+        "kde4-config"
+        "kfile"
+        "kmimetypefinder"
+        "kmimetypefinder5"
+        "ktraderclient"
+        "ktradertest"
+        "mimetype"
+        "qtpaths"
+        "qtxdg-mat"
+      ];
+      fix."/usr/bin/file" = true;
+      keep = {
+        "$KDE_SESSION_VERSION" = true;
+        "$KTRADER" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-open" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ nettools "${placeholder "out"}/bin" ];
+      execer = [
+        "cannot:${placeholder "out"}/bin/xdg-mime"
+      ];
+      fake.external = commonFakes ++ [
+        "cygstart"
+        "dde-open"
+        "enlightenment_open"
+        "env"
+        "exo-open"
+        "gdbus"
+        "gio"
+        "gnome-open"
+        "gvfs-open"
+        "kde-open"
+        "kfmclient"
+        "mate-open"
+        "mimeopen"
+        "open"
+        "pcmanfm"
+        "qtxdg-mat"
+        "run-mailcap"
+        "rundll32.exe"
+        "wslpath"
+      ];
+      keep = {
+        "$browser" = true;
+        "$KDE_SESSION_VERSION" = true;
+        "$printf" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-screensaver" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ nettools perl procmail procps ];
```

Comment:
I'm not sure if xdg-utils is widely-used enough on macOS for this to actually be a blocker, but when testing this I noticed that procmail isn't currently building on macOS. Since it's added as a dependency here, it'll effectively mean breaking xdg-utils on mac until that's also fixed. 

I opened an issue for this build failure: https://github.com/NixOS/nixpkgs/issues/286169 (otherwise haven't looked into it; trying to stay focused on this review :)

Reply:
Sure hope it's not.